src/main/java/luke/Luke.java

```diff
@@ -0,0 +1,301 @@
+package luke;
+
+import luke.exceptions.IncorrectInput;
+import luke.exceptions.InsufficientArguments;
+import luke.exceptions.InvalidCommand;
+import luke.exceptions.LukeException;
+import luke.tasks.Deadline;
+import luke.tasks.Event;
+import luke.tasks.Task;
+import luke.tasks.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class Luke {
+
+    private static final String LOGO =
+            "                              \n" +
+            ",--.           ,--.           \n" +
+            "|  |   ,--.,--.|  |,-. ,---.  \n" +
+            "|  |   |  ||  ||     /| .-. : \n" +
+            "|  '--.'  ''  '|  \\  \\\\   --. \n" +
+            "`-----' `----' `--'`--'`----' \n" +
+            "                              ";
+
+    private static final String HORIZONTAL_LINE =
+            "____________________________________________________________";
+
+    public static final int MAX_TASK_COUNT = 100;
+    private static final int TASK_TYPE_INDEX = 0;
+    private static final int ISDONE_INDEX = 1;
+    private static final int DESCRIPTION_INDEX = 2;
+    private static final int BY_INDEX = 3;
+    private static final int FROM_INDEX = 3;
+    private static final int TO_INDEX = 4;
+    private static ArrayList<Task> tasks = new ArrayList<>();
+//    private static Task[] tasks = new Task[MAX_TASK_COUNT];
+//    private static int size = 0;
+
+    private static void printReply(String reply) {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println(reply);
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static void printDivider() {
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    private static String numberOfTasksMessage() {
+        return String.format("Now you have %d %s in the list.", tasks.size(), tasks.size() > 1 ? "tasks" : "task");
+    }
+
+    private static void list() {
+        printDivider();
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.printf("%d. ", i + 1);
+            System.out.println(tasks.get(i).toString());
+        }
+        printDivider();
+    }
+
+    private static void saveAllTasks() throws IOException {
+        FileWriter fw = new FileWriter("data/tasks.txt");
+        String line;
+        for (Task t : tasks) {
+            line = t.toString2();
+            fw.write(line);
+            fw.write(System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    private static void executeCommand(CommandType commandType, String[] inputArr) {
```

Comment:
This is a very long method. Perhaps extract functionalities out for each command so as to allow your code to be more readable?